include/deal.II/matrix_free/evaluation_kernels.h

```diff
@@ -376,7 +376,8 @@ namespace internal
                                    : (Eval::dofs_per_cell > Eval::n_q_points ?
                                       Eval::dofs_per_cell : Eval::n_q_points);
     const unsigned int max_stack_size = 100;
-    VectorizedArray<Number>  temp_data[(temp_size > 0 && temp_size < max_stack_size) ? 2*temp_size : 1];
+    AlignedVector<VectorizedArray<Number>>  temp_data;
+    temp_data.resize_fast((temp_size > 0 && temp_size < max_stack_size) ? 2*temp_size : 1);
```

Comment:
You should not allocate memory in this function. This is way too expensive here. `temp_data` was meant to be used as a short array on the stack to allow the compiler to gather more aliasing information and avoid storing some temporary results back to the scratch array in case it manages to hold the data in registers. But this function is getting less important anyway because one of my next steps for DG after #4680 is to commit the evaluators that also use the transformation to the collocation space in all cases with `n_q_points_1d >= fe_degree+1` which do not need the temporary data array. So my suggestion is to skip `temp_data` completely and only use the else case 20 lines below that sets `temp1=scratch_data` and `temp2=temp1+temp_size`. That should also avoid the compiler warning. I have seen those on some compiler for some configuration too, but it usually appeared either when I had a bug in the tensor product evaluators or when there were other compile errors in which case some compilers spit out other warnings because they did not finish all analysis passes...